File: src/entity/tile/tiles/SpawnTeamTile.java
package entity.tile.tiles;

import misc.game.Team;
import entity.tile.Tile;

public class SpawnTeamTile extends Tile
{
	private Team team;

	public SpawnTeamTile(Team team)
	{
		setTeam(team);
	}

	// Setter
	public void setTeam(Team team) { this.team = team; }

	// Getter
	public Team getTeam() { return team; }
}

File: cpp/src/entity/tiles/SpawnTeamTile.cpp
#include "SpawnTeamTile.hpp"

#include <graphics/GraphicsManager.hpp>

SpawnTeamTile::SpawnTeamTile(Team* team, const GameVector& pos)
	: Tile(pos), team(team)
{}

sf::Texture* SpawnTeamTile::getTexture() const
{
	return GraphicsManager::getTexture(GraphicsIDs::SPAWNTEAMTILE);
}

bool SpawnTeamTile::isSpawnTeamTile() const
{
	return true;
}

Team* SpawnTeamTile::getTeam() const
{
	return team;
}

std::string SpawnTeamTile::toString() const
{
	return "SpawnTeamTile";
}

float SpawnTeamTile::getCollisionPriority(Entity* e) const
{
	return -1000;
}
